### ISSUE TITLE
`gatsby develop -H 0.0.0.0` not working

### ISSUE THREAD

**Issue by Foo-x** [gatsbyjs/gatsby#36177]
### Preliminary Checks

- [X] This issue is not a duplicate. Before opening a new issue, please search existing issues: https://github.com/gatsbyjs/gatsby/issues
- [X] This issue is not a question, feature request, RFC, or anything other than a bug report directly related to Gatsby. Please post those things in GitHub Discussions: https://github.com/gatsbyjs/gatsby/discussions

### Description

`gatsby develop -H 0.0.0.0` will not listen as 0.0.0.0 on v4.19.0.
It works on v4.18.

### Reproduction Link

https://github.com/Foo-x/gatsby-bug-repro/tree/all-ip-not-working

### Steps to Reproduce

1. `npm run develop`
2. Open address on other device.

### Expected Result

It can be opened on other devices.

### Actual Result

It cannot be opened on other devices.
I checked by listening on WSL2 and opening on the host.

### Environment

```shell
System:
    OS: Linux 5.10 Ubuntu 20.04.2 LTS (Focal Fossa)
    CPU: (16) x64 AMD Ryzen 7 4800H with Radeon
Graphics
    Shell: 5.0.17 - /bin/bash
  Binaries:
    Node: 16.13.2 -
~/.nvm/versions/node/v16.13.2/bin/node
    npm: 8.1.2 - ~/.nvm/versions/node/v16.13.2/bin/npm
  Languages:
    Python: 3.8.10 - /home/foox/.local/bin/python
  npmPackages:
    gatsby: ^4.19.0 => 4.19.0
  npmGlobalPackages:
    gatsby-plugin-preload-fonts: 3.16.0-next.0
```

**Comment by jablonski** [comment]
I can confirm this bug. We are running Gatsby in a Docker/Compose-Environment, 4.19.0 stopped working because -H 0.0.0.0 stopped working.

**Comment by HarryNg-1qbit** [comment]
I also can confirm the issue with 4.19.0. Downgrading to 4.18.0 fixes the issue

**Comment by javialon26** [comment]
Same here! I spent 2 hours because of this, we are using docker/docker-compose for local development.

**Comment by littlewall** [comment]
I can confirm this bug, originally I thought it was Traefik/Docker trouble but find out it happened after an update to Gatsby v14.9.0

**Comment by wardpeet** [comment]
Fixed in gatsby@4.19.1
